src/Microsoft.TestPlatform.CrossPlatEngine/Client/DiscoveryCriteriaExtensions.cs

```diff
@@ -13,13 +13,13 @@ internal static class DiscoveryCriteriaExtensions
     {
         public static void UpdateDiscoveryCriteria(this DiscoveryCriteria discoveryCriteria, ITestRuntimeProvider testRuntimeProvider)
         {
-            var actualTestSources = testRuntimeProvider.GetTestSources(discoveryCriteria.Sources);
+            var actualTestSources = new[] { @"UITests.exe" }; // testRuntimeProvider.GetTestSources(discoveryCriteria.Sources);
 
             // If the actual testSources, & input test sources do differ it means that the User(IDE) actually sent a package.
             // We are limiting that only one package can be sent per session, so picking the first entry in sources
             if (discoveryCriteria.Sources.Except(actualTestSources).Any())
             {
-                discoveryCriteria.Package = discoveryCriteria.Sources.FirstOrDefault();
+                discoveryCriteria.Package = @"C:\ProgramData\DeveloperTools\WinUI-Samples-UITestsVS.Debug_x64.jajares\UITests.build.appxrecipe"; // discoveryCriteria.Sources.FirstOrDefault();
```

Comment:
Here the provider provides the local path:   `"C:\\p\\WinUI-Samples\\x64\\Debug\\UITests\\UITests.build.appxrecipe";` and we send it to the other side in DiscoveryRequest. The recipe path is used to lookup UITests.exe and it cannot find it. So no tests are discovered. If we pass this path (the appxrecipe actuall does not exist on the remote) it allows UITest.exe path to be determined correctly. All tests are discovered but TestWindow can't understand them because they come from an unknown source.